src/render/src/device.rs

```diff
@@ -82,10 +76,19 @@ impl<S> From<CreationError> for PipelineStateError<S> {
         PipelineStateError::DeviceCreate(e)
     }
 }
+*/
+
+#[derive(Clone)]
+pub struct Device<B: Backend> {
+    pub(crate) device: B::Device,
+    // TODO: could be shared instead of cloned
+    pub(crate) heap_types: Vec<HeapType>,
```

Comment:
why do all of them need to be `pub(crate)`?

Reply:
It was mainly to avoid writing a redundant `Device::new` but I can remove them.